src/program.js

```diff
@@ -65,54 +65,52 @@ export class Program {
     return this;
   }
 
-  run(absolutePackageDir: string,
+  async run(absolutePackageDir: string,
       {systemProcess=process, logStream=defaultLogStream,
        getVersion=defaultVersionGetter, shouldExitProgram=true}
-      : Object = {}): Promise<any> {
+      : Object = {}): Promise<void> {
 
     this.shouldExitProgram = shouldExitProgram;
     let argv;
     let cmd;
 
-    return new Promise(
-      (resolve) => {
-        this.yargs.exitProcess(this.shouldExitProgram);
-        argv = this.yargs.argv;
+    this.yargs.exitProcess(this.shouldExitProgram);
+    argv = this.yargs.argv;
```

Comment:
I think this can just be `let argv = ...` since we no longer have to expose it to nested promise callback scopes.